cmd/keeper/keeper.go

```diff
@@ -1754,3 +1757,94 @@ func keeper(c *cobra.Command, args []string) {
 
 	lockFile.Close()
 }
+
+// IsMaster return if the db is the cluster master db.
+// A master is a db that:
+// * Has a master db role
+// or
+// * Has a standby db role with followtype external
+func IsMaster(db *cluster.DB) bool {
+	switch db.Spec.Role {
+	case common.RoleMaster:
+		return true
+	case common.RoleStandby:
+		if db.Spec.FollowConfig.Type == cluster.FollowTypeExternal {
+			return true
+		}
+		return false
+	default:
+		panic("invalid db role in db Spec")
+	}
+}
+
+func (p *PostgresKeeper) generateHBA(cd *cluster.ClusterData, role common.Role, userHba []string) []string {
+
+	// get keeper current db, current db should not be nil
+	k, ok := cd.Keepers[p.keeperLocalState.UID]
+	if !ok {
+		panic(fmt.Sprintf("Unable to get Keeper data with keeper UID %s", p.keeperLocalState.UID))
+	}
+	currentDb := cd.FindDB(k)
+	if currentDb == nil {
+		panic(fmt.Sprintf("Unable to get current keeper local DB with UID %s", p.keeperLocalState.UID))
+	}
+
```

Comment:
* you already have the `db` in the caller just pass it to generateHBA without all of the above (and avoid calls to panic)
* `role common.Role` isn't used anymore
* rename `userHba` to `specHBA` to be clearer.

generateHBA should become:
```
func (p *PostgresKeeper) generateHBA(cd *cluster.ClusterData, db *cluster.DB, specHBA []string) []string {
```

Reply:
Ok but since I get specHBA from `db.spec.PGHBA` I could simplify and just only call `func (p *PostgresKeeper) generateHBA(cd *cluster.ClusterData, db *cluster.DB) []string` no ?

Reply:
right!